demo/admin/src/news/generated/NewsGrid.tsx

```diff
@@ -0,0 +1,235 @@
+        {
+            field: "actions",
+            headerName: "",
+            sortable: false,
+            filterable: false,
+            type: "actions",
+            renderCell: (params) => {
+                return (
+                    <>
+                        <IconButton component={StackLink} pageName="edit" payload={params.row.id}>
+                            <Edit color="primary" />
+                        </IconButton>
+                        <CrudContextMenu
+                            copyData={() => {
+                                const row = params.row;
+                                return {
+                                    category: row.category,
+                                    content: NewsContentBlock.state2Output(NewsContentBlock.input2State(row.content)),
+                                    date: row.date,
+                                    image: DamImageBlock.state2Output(DamImageBlock.input2State(row.image)),
+                                    slug: row.slug,
+                                    title: row.title,
+                                };
+                            }}
+                            onPaste={async ({ input }) => {
+                                await client.mutate<GQLCreateNewsMutation, GQLCreateNewsMutationVariables>({
+                                    mutation: createNewsMutation,
+                                    variables: { scope, input },
+                                });
+                            }}
+                            onDelete={async () => {
+                                await client.mutate<GQLDeleteNewsMutation, GQLDeleteNewsMutationVariables>({
+                                    mutation: deleteNewsMutation,
+                                    variables: { id: params.row.id },
+                                });
+                            }}
+                            refetchQueries={[newsQuery]}
+                            // url={url}
+                        />
+                    </>
+                );
+            },
+        },
+    ];
+
+    const { filter: gqlFilter, search: gqlSearch } = muiGridFilterToGql(columns, dataGridProps.filterModel);
+
+    const { data, loading, error } = useQuery<GQLNewsGridQuery, GQLNewsGridQueryVariables>(newsQuery, {
+        variables: {
+            scope,
+            filter: gqlFilter,
+            search: gqlSearch,
+            offset: dataGridProps.page * dataGridProps.pageSize,
+            limit: dataGridProps.pageSize,
+            sort: muiGridSortToGql(dataGridProps.sortModel),
+        },
+    });
+    if (error) throw error;
+    const rows = data?.newsList.nodes ?? [];
+    const rowCount = useBufferedRowCount(data?.newsList.totalCount);
+
+    return (
+        <Box sx={{ height: `calc(100vh - var(--comet-admin-master-layout-content-top-spacing))` }}>
```

Comment:
Out of scope: We need to find a better solution for this

Reply:
yes

Reply:
Not sure if this has been merged into `next` yet, but if it is, you can use `<MainContent />` with the `fullHeight` prop here: https://github.com/vivid-planet/comet/pull/1221